internal/typeinfo/info.go

```diff
@@ -0,0 +1,34 @@
+package typeinfo
+
+import (
+	"reflect"
+)
+
+// Field represents a single field from a struct type.
+type Field struct {
+	Type reflect.Type
+
+	// Name is the name of the struct field.
+	Name string
+
+	// Index of this field in the structure.
+	Index int
+
+	// OmitEmpty is true when "omitempty" is
+	// a property of the field's "db" tag.
+	OmitEmpty bool
+}
+
+// Info represents reflected information about a struct type.
+type Info struct {
+	Type reflect.Type
+
+	// Relate tag names to fields.
+	TagToField map[string]Field
+
+	// Relate field names to tags.
+	FieldToTag map[string]string
+}
+
+// SQLAir's specific map type.
+type M map[string]any
```

Comment:
This can go now, I believe. There's no use for it inside the package for the time being.